keras_core/callbacks/backup_and_restore_callback.py

```diff
@@ -0,0 +1,198 @@
+    recently saved state at the beginning of a new Model.fit run. At the
+    completion of a Model.fit run, the temporary checkpoint file is deleted.
+
+    Note that the user is responsible to bring jobs back after the
+    interruption. This callback is important for the backup and restore
+    mechanism for fault tolerance purpose, and the model to be restored from
+    a previous checkpoint is expected to be the same as the one used to back
+    up. If user changes arguments passed to compile or fit, the checkpoint
+    saved for fault tolerance can become invalid.
+
+    Args:
+        backup_dir: String, path to store the checkpoint. e.g. backup_dir =
+            os.path.join(working_dir, 'backup'). This is the directory in which
+            the system stores temporary files to recover the model from jobs
+            terminated unexpectedly.
+        save_freq: 'epoch', integer, or False. When set to 'epoch'
+            the callback saves the checkpoint at the end of each epoch. When set
+            to an integer, the callback saves the checkpoint every save_freq
+            batches. Set save_freq to False if only using preemption
+            checkpointing (with save_before_preemption=True).
+        delete_checkpoint: Boolean, default to True. This BackupAndRestore
+            callback works by saving a checkpoint to back up the training state.
+            If delete_checkpoint=True, the checkpoint will be deleted after
+            training is finished. Use False if you'd like to keep the checkpoint
+            for future usage.
+        save_before_preemption: A boolean value instructing whether to turn on
+            the automatic checkpoint saving for preemption/maintenance events.
+
+    """
+
+    def __init__(
+        self,
+        backup_dir,
+        save_freq="epoch",
+        delete_checkpoint=True,
+        save_before_preemption=False,
+    ):
+        super().__init__()
+        self._current_epoch = 0
+        self.save_freq = save_freq
+        self.delete_checkpoint = delete_checkpoint
+        self.save_before_preemption = save_before_preemption
+        self._last_batch_seen = 0
+
+        if not backup_dir:
+            raise ValueError("Empty `backup_dir` argument passed")
+        self.backup_dir = backup_dir
+
+        if (not save_freq) and (not save_before_preemption):
+            raise ValueError(
+                "Either `save_freq` or `save_before_preemption` " "must be set."
+            )
+
+        if self.save_freq != "epoch" and not isinstance(self.save_freq, int):
+            raise ValueError(
+                f"Unrecognized save_freq: {self.save_freq}. "
+                "Expected save_freq are 'epoch' or integer values"
+            )
+
+    def on_train_begin(self, logs=None):
+        """
+        Get training state from temporary file and restore it
+        """
+        super().on_train_begin()
+        self.set_model(keras_core.saving.load_model(filepath=self.backup_dir))
```

Comment:
Is this the right approach? Should we rather take `self.model` and load its state?

Reply:
Not able to directly use **self.model** but **self._model** seems to be working